pkg/reconciler/pipeline/controller.go

```diff
@@ -127,6 +128,9 @@ func (r *pipelineReconciler) reconcile(ctx context.Context, pl *dfv1.Pipeline) (
 			controllerutil.RemoveFinalizer(pl, finalizerName)
 			// Clean up metrics
 			_ = reconciler.PipelineHealth.DeleteLabelValues(pl.Namespace, pl.Name)
+			// Delete corresponding vertex metrics
+			_ = reconciler.VertexDisiredReplicas.DeletePartialMatch(map[string]string{metrics.LabelNamespace: pl.Namespace, metrics.LabelPipeline: pl.Name})
```

Comment:
what does `DeletePartialMatch` do?

Reply:
Delete the metrics with matching labels - do not need to explicitly to delete from each vertex.